classifiers/classifier_images.py

```diff
@@ -162,15 +162,16 @@ def val_gzsl(self, test_X, test_label, target_classes):
         predicted_label = torch.LongTensor(test_label.size())
         for i in range(0, ntest, self.batch_size):
             end = min(ntest, start+self.batch_size)
-            if self.cuda:
-                # inputX = Variable(test_X[start:end].cuda(), volatile=True)
-                inputX = test_X[start:end].detach().cuda()
-            else:
-                # inputX = Variable(test_X[start:end], volatile=True)
-                inputX = test_X[start:end].detach()
-            output = self.model(inputX)  
-            _, predicted_label[start:end] = torch.max(output.data, 1)
-            start = end
+            with torch.no_grad():
+                if self.cuda:
+                    # inputX = Variable(test_X[start:end].cuda(), volatile=True)
```

Comment:
During testing, make sure you use the following correct code.
```
model.eval()
with torch.no_grad():
    pred = model(data)
model.train()
```

Reply:
Okay. I will check it again.